src/Microsoft.HttpRepl.Tests/OpenApi/SwaggerV1ApiDefinitionReaderTests.cs

```diff
@@ -0,0 +1,358 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.HttpRepl.OpenApi;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.HttpRepl.Tests.OpenApi
+{
+    public class SwaggerV1ApiDefinitionReaderTests
+    {
+        [Fact]
+        public void ReadMetadata_WithNoApis_ReturnsApiDefinitionWithNoDirectories()
+        {
+            string json = @"{
+  ""swaggerVersion"": ""1.0.0"",
+  ""info"": {
+    ""version"": ""v1""
+  }
+}";
+            JObject jobject = JObject.Parse(json);
+            SwaggerV1ApiDefinitionReader swaggerV1ApiDefinitionReader = new SwaggerV1ApiDefinitionReader();
+
+            ApiDefinition apiDefinition = swaggerV1ApiDefinitionReader.ReadDefinition(jobject, null);
+
+            Assert.NotNull(apiDefinition?.DirectoryStructure);
+            Assert.Empty(apiDefinition.DirectoryStructure.DirectoryNames);
+        }
+
+        [Fact]
+        public void ReadMetadata_WithNoPath_ReturnsApiDefinitionWithNoDirectories()
+        {
+            string json = @"{
+  ""swaggerVersion"": ""1.0.0"",
+  ""apis"": [
+    {
+        ""description"": ""resource1""
+    }]
+}";
+            JObject jobject = JObject.Parse(json);
+            SwaggerV1ApiDefinitionReader swaggerV1ApiDefinitionReader = new SwaggerV1ApiDefinitionReader();
+
+            ApiDefinition apiDefinition = swaggerV1ApiDefinitionReader.ReadDefinition(jobject, null);
+
+            Assert.NotNull(apiDefinition?.DirectoryStructure);
+            Assert.Empty(apiDefinition.DirectoryStructure.DirectoryNames);
+        }
+
+        [Fact]
+        public void ReadMetadata_WithNoMethods_ReturnsApiDefinitionWithNullRequestInfo()
+        {
+            string json = @"{
+  ""swaggerVersion"": ""1.2"",
+  ""apis"": [
+    {
+      ""path"": ""/user/logout"",
+      ""operations"": [
+        {
+        }
+      ]
+    }
+  ]
+}";
+            JObject jobject = JObject.Parse(json);
+            SwaggerV1ApiDefinitionReader swaggerV1ApiDefinitionReader = new SwaggerV1ApiDefinitionReader();
+
+            ApiDefinition apiDefinition = swaggerV1ApiDefinitionReader.ReadDefinition(jobject, null);
+
+            Assert.NotNull(apiDefinition?.DirectoryStructure);
```

Comment:
Is this also a pre-assert?  Isn't this tested elsewhere in the overall suite?

Reply:
It's probably more accurate to say that I was just testing two different things here (without methods in the swagger file it should still return a directory structure AND without methods in the swagger file, that directory structure should have no request info). I split it up into separate tests. I could definitely see a bug happening where when there are no methods we bail out and don't set the structure. That wasn't tested elsewhere, so I included it.